src/jaxsim/api/data.py

```diff
@@ -625,9 +624,8 @@ def reset_base_pose(self, base_pose: jtp.MatrixLike) -> Self:
 
         W_p_B = base_pose[0:3, 3]
 
-        to_wxyz = np.array([3, 0, 1, 2])
-        W_R_B: jaxlie.SO3 = jaxlie.SO3.from_matrix(base_pose[0:3, 0:3])  # noqa
-        W_Q_B = W_R_B.as_quaternion_xyzw()[to_wxyz]
+        W_R_B: jaxlie.SO3 = jaxlie.SO3.from_matrix(base_pose[0:3, 0:3])
+        W_Q_B = W_R_B.wxyz
```

Comment:
Can we use `jaxsim.math.Quaternion` here?

Reply:
Do you mean something like `W_Q_B = jaxsim.math.Quaternion.from_dcm(dcm=base_pose[:3][:3])`?

Reply:
Yep, when possible I'd like to remove direct usage of `jaxlie` in favor of `jaxsim.math.*` resources. 

```suggestion
        W_Q_B = jaxsim.math.Quaternion.from_dcm(dcm=base_pose[0:3, 0:3])
```